### ISSUE TITLE
archivejson not returning the proper data when count= and start= used

### ISSUE THREAD

**Issue by ssaxnagios** [NagiosEnterprises/nagioscore#705]
We were able to reproduce in older versions as well with archivejson and the count/data.

Here is where that occurs:

https://github.com/NagiosEnterprises/nagioscore/blob/345fd4e8257085ce7f3806ee4fb04008919f24e5/cgi/jsonutils.c#L916
Looks like you can set &start=0 to start at index 0:

http://<nagios_server>/nagios/cgi-bin/archivejson.cgi?query=alertlist&count=10&dateformat=%25d%2F%25m+%25H%3A%25M&starttime=-86400&endtime=%2B0&hoststates=down+unreachable&servicestates=critical+unknown&statetypes=hard&start=0

I'm still having the same issue after adding that variable. Let me give you an example:

1) Setting the count value to 10, this is the last entry for the alertlist array in the generated JSON

{
        "timestamp": "07/10 22:22",
        "object_type": 2,
        "host_name": "XXXXX",
        "description": "Ping XXXXX",
        "state_type": 1,
        "state": 32,
        "plugin_output": "Critico - XXXXX- Perda de pacote: XXXXXXX (60%) (33.0) ms"
      }

2) If I set count to 11, these are the last 2 entries:

{
        "timestamp": "07/10 22:22",
        "object_type": 2,
        "host_name": "XXXXX",
        "description": "Ping XXXXX",
        "state_type": 1,
        "state": 32,
        "plugin_output": "Critico - Alcantara  - Perda de pacote: XXXXXXX (60%) (33.0) ms"
      },
      {
        "timestamp": "07/10 23:02",
        "object_type": 2,
        "host_name": "XXXXX",
        "description": "Ping XXXXX",
        "state_type": 1,
        "state": 32,
        "plugin_output": "Critico - XXXXX - Perda de pacote: XXXXXXX  (40%) (65.8) ms"

As seen above, the 07/10 22:22 timestamp is not the last entry in the archives, but is being omitted because the count value is not reaching the number of entries stored in the time frame specified (-86400).

But that is exactly what I want! The issue I've found is not starting from the last entry available and going down from there - being limited by the count option's range.

As seen above, the 07/10 22:22 timestamp is not the last entry in the archives, but the next entry is being omitted because the count value is not reaching the number of entries stored in the time frame specified (-86400).

**Comment by sawolf** [comment]
I'm not very familiar with the archivejson cgi, but what's the issue here? It seems like the first half of this issue is missing. From what's actually written here, the user asks for 10 entries, gets <whatever> (I assume the first 9 entries were omitted), asks for 11 entries, and gets one more.

**Comment by ssaxnagios** [comment]
I think I may have misread this.

In order for this to work you WOULD NOT use the start=0 since that means start at the top of the array.

@sawolf I don't think the default behavior is meant to read the array in reverse, can you confirm? In addition to that, I think the start option is meant to be used as an offset from the default array processing of index 0 and let's you specify a higher index (than the default 0) to skip entries in the list not to start at the other end of the array, correct?

**Comment by sawolf** [comment]
Yeah, it looks like the current functionality for archivejson.cgi is that it collects items starting at starttime, ends either at endtime or at start+count items, and returns that JSON. Essentially it's giving you filtering options to paginate your results in a way that starts at the earliest time and ends at the most recent time. I'll agree with the customer that it's somewhat silly how there aren't any sorting options, but they'll need to work around it for now.
